sycl/doc/CompilerAndRuntimeDesign.md

```diff
@@ -819,29 +820,27 @@ Depending on the compiler mode `multi_ptr` will either annotate internal data
 with address space attribute or not.
 
 To utilize existing clang's functionality, we re-use following OpenCL address
-space attributes in SYCL mode with modified semantic rules and unmodified parser
-and IR generation components.
-
-``` C++
-  __attribute__((opencl_global))
-  __attribute__((opencl_local))
-  __attribute__((opencl_private))
-```
-
-The main difference in Sema component with OpenCL mode is that SYCL mode
-(similar to other single-source GPU programming modes like OpenMP/CUDA/HIP)
-keeps "default" address space for the declaration without explicit address space
-attribute annotations. This keeps the type system for the code shared between
-the host and device semantically-equivalent for both compilers: regular C++ host
-compiler and SYCL compiler.
-
-SYCL mode allows conversion from annotated pointer to unannotated pointer to
-enable integration of accelerated code with standard C++ code.
-
-> **NOTE:** casting unannotated pointer to address-space annotated pointer must
-be done with care. If address space of annotated pointer doesn't match the
-address space of the allocation unannotated pointer is pointing to, the behavior
-of cast operation is undefined.
+space attributes in SYCL mode: `__attribute__((opencl_global))`,
+`__attribute__((opencl_global_host))`, `__attribute__((opencl_global_device))`,
+`__attribute__((opencl_local))`, `__attribute__((opencl_private))` and
+`__attribute__((opencl_constant))`.
+
+The main address space semantic difference of SYCL mode from OpenCL is that
+SYCL doesn't assign OpenCL generic address space to a declaration's type without explicit
+address space attribute. Similar to other single-source C++-based GPU
+programming modes like OpenMP/CUDA/HIP, SYCL uses
+clang's "default" address space for types with no address space attributes.
+During the lowering to LLVM IR, the default address space is mapped to
+the SPIR generic address space. Declarations are assigned to the
+relevant memory region depending on their declaration context and
+pointers to them are cast to generic. This design has two important features: keeps
```

Comment:
> Declarations are assigned to the
attribute annotations. This keeps the type system for the code shared between	relevant memory region depending on their declaration context and
the host and device semantically-equivalent for both compilers: regular C++ host	pointers to them are cast to generic.

Not quite clear. Maybe give an example?

Reply:
Here is @Naghasan's comment with the example: https://reviews.llvm.org/D80932#2073542.
Do you suggest adding something similar to this document?

Reply:
Yes, please. I can understand that text with examples much better.

Reply:
Ok. I'll steal Victor's example from https://reviews.llvm.org/D80932#2073542. @Naghasan, I hope you don't mind.